qgym/utils/qiskit_utils.py

```diff
@@ -0,0 +1,96 @@
+"""This module contains utility functions for interactions with the qiskit library."""
+
+from __future__ import annotations
+
+from collections.abc import Hashable
+
+import networkx as nx
+import numpy as np
+from numpy.typing import NDArray
+from qiskit import QuantumCircuit
+from qiskit.converters import circuit_to_dag
+from qiskit.dagcircuit import DAGCircuit
+from qiskit.transpiler import Layout
+
+
+def get_interaction_graph(circuit: QuantumCircuit | DAGCircuit) -> nx.Graph:
+    """Create and interaction graph from the provided `circuit`.
+
+    Args:
+        circuit: Circuit to produce the interactions from. Both
+            :class:`~qiskit.circuit.QuantumCircuit` and
+            :class:`~qiskit.dagcircuit.DAGCircuit` representations are accepted.
+
+    Returns:
+        Interaction graph of the `circuit`.
+    """
+    dag = _parse_circuit(circuit)
+
+    if dag.multi_qubit_ops():
+        msg = "no 3+ qubit operations are supported"
+        raise ValueError(msg)
+
+    qreg_to_int = _get_qreg_to_int_mapping(dag)
+    interaction_graph: nx.Graph = nx.empty_graph(dag.num_qubits())
+
+    for op_node in dag.op_nodes(include_directives=False):
+        if len(op_node.qargs) == 1:
+            continue
+        qubit1 = qreg_to_int[op_node.qargs[0]]
+        qubit2 = qreg_to_int[op_node.qargs[1]]
+        interaction_graph.add_edge(qubit1, qubit2)
```

Comment:
This can probably be writen as:

```
interaction_graph.add_nodes_from([
    ( qreg_to_int[op_node.qargs[0]], qreg_to_int[op_node.qargs[1]] )
    for op_node in dag.op_nodes(include_directives=False) if len(op_node.qargs) == 1
])
```

Reply:
Yes, it's rewritten to use the comprehension

Reply:
Rewritten, and properly, using `add_edges_from` and not `add_nodes_from`. Thanks!